business/MetadataDatabase/Models/QidoSeries.cs

```diff
@@ -0,0 +1,155 @@
+
+        // Primary hospital identification number or code for the patient. (00100020)
+        [JsonPropertyName("00100020")]
+        public DicomStringObject PatientID { get; set; }
+
+        // Birth date of the patient. (00100030)
+        [JsonPropertyName("00100030")]
+        public DicomStringObject PatientsBirthDate { get; set; }
+
+        // Sex of the named patient. (00100040)
+        [JsonPropertyName("00100040")]
+        public DicomStringObject PatientsSex { get; set; }
+
+        // Unique identifier for the Study (0020000D)
+        [JsonPropertyName("0020000D")]
+        public DicomStringObject StudyInstanceUID { get; set; }
+
+        // Unique identifier of the Series. (0020000E)
+        [JsonPropertyName("0020000E")]
+        public DicomStringObject SeriesInstanceUID { get; set; }
+
+        // User or equipment generated Study identifier (00200010)
+        [JsonPropertyName("00200010")]
+        public DicomStringObject StudyID { get; set; }
+
+        // A number that identifies this Series. (00200011)
+        [JsonPropertyName("00200011")]
+        public DicomIntObject SeriesNumber { get; set; }
+
+        // The number of Composite Object Instances in a Series that match the Series level Query/Retrieve search criteria (00201209)
+        [JsonPropertyName("00201209")]
+        public DicomIntObject NumberOfSeriesRelatedInstances { get; set; }
+
+        // Todo
+        public string GetValueOfDicomTag(DicomTag propertyName)
+        {
+            string resultValue = "";
+            var dicomPrpertyType = this.GetType().GetProperty(propertyName.Value).PropertyType.Name;
+            switch (dicomPrpertyType)
+            {
+                case nameof(DicomStringObject):
+                    var dicomStringValue = (DicomStringObject)this.GetType().GetProperty(propertyName.Value).GetValue(this, null);
+                    if (dicomStringValue.Value != null)
+                    {
+                        resultValue = dicomStringValue.Value[0];
+                    }
+                    break;
+                case nameof(DicomIntObject):
+                    var dicomIntValue = (DicomIntObject)this.GetType().GetProperty(propertyName.Value).GetValue(this, null);
+                    if (dicomIntValue.Value != null)
+                    {
+                        resultValue = dicomIntValue.Value[0].ToString();
+                    }
+                    break;
+                case nameof(DicomNameObject):
+                    var dicomNameObject = (DicomNameObject)this.GetType().GetProperty(propertyName.Value).GetValue(this, null);
+                    if (dicomNameObject.Value != null)
+                    {
+                        resultValue = dicomNameObject.Value[0].Alphabetic;
+                    }
+                    break;
+                default:
+                    break;
+            }
+            return resultValue;
+        }
+
+        public class DicomTag
```

Comment:
move to dedicated file

Reply:
Replace this by en enum !
in C#, myEnum.MyEnumeValue123.ToString() returns "MyEnumeValue123"